### ISSUE TITLE
Pass phrase strength should be properly checked before encrypting

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-android#606]
```
[STACK_TRACE]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
        at com.flowcrypt.email.util.exception.ExceptionUtil$Companion.handleError(ExceptionUtil.kt:30)
        at com.flowcrypt.email.service.actionqueue.ActionQueueIntentService.onHandleWork(ActionQueueIntentService.kt:59)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:392)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
Caused by: com.flowcrypt.email.util.exception.NodeException: Error :Error: Pass phrase length seems way too low! Pass phrase strength should be properly checked before encrypting a key.
Stack :Error: Pass phrase length seems way too low! Pass phrase strength should be properly checked before encrypting a key.
    at Endpoints.encryptKey ([eval]:71873:15)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Type :null
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.checkResult(NodeCallsExecutor.kt:189)
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.encryptKey(NodeCallsExecutor.kt:134)
        at com.flowcrypt.email.service.actionqueue.actions.EncryptPrivateKeysIfNeededAction.run(EncryptPrivateKeysIfNeededAction.kt:73)
        at com.flowcrypt.email.service.actionqueue.ActionQueueIntentService.onHandleWork(ActionQueueIntentService.kt:53)
        ... 7 more
com.flowcrypt.email.util.exception.NodeException: Error :Error: Pass phrase length seems way too low! Pass phrase strength should be properly checked before encrypting a key.
Stack :Error: Pass phrase length seems way too low! Pass phrase strength should be properly checked before encrypting a key.
    at Endpoints.encryptKey ([eval]:71873:15)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Type :null
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.checkResult(NodeCallsExecutor.kt:189)
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.encryptKey(NodeCallsExecutor.kt:134)
        at com.flowcrypt.email.service.actionqueue.actions.EncryptPrivateKeysIfNeededAction.run(EncryptPrivateKeysIfNeededAction.kt:73)
        at com.flowcrypt.email.service.actionqueue.ActionQueueIntentService.onHandleWork(ActionQueueIntentService.kt:53)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:392)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
```

This error is thrown on purpose when passphrase seems very unreasonably short (< 12 characters). Such pass phrases should not have passed the zxcvbn check, so it's possible that they were not always checked (eg already imported a weaker pass phrase, then trying to back up or something like that).

**Comment by DenBond7** [comment]
ref https://mail.google.com/mail/u/0/#search/%22Pass+phrase+length+seems+way+too+low%22

**Comment by tomholub** [comment]
From the emails it looks like it's happening repeatedly. Is it re-trying? Or is each error user-initiated?

**Comment by DenBond7** [comment]
@tomholub Such errors happen when the app is trying to run a background task to encrypt keys which were wrongly saved (not encrypted - we had a bug when we were creating a key  https://github.com/FlowCrypt/flowcrypt-android/issues/447). Now when the app is starting it's trying to encrypt that keys in the local database ---> the error is happening --> a new task is scheduling --> when the app is starting we have the same logic. Users don't see any errors but we receive ACRA reports

**Comment by tomholub** [comment]
You could cancel the background task when no pass phrase is available.

**Comment by DenBond7** [comment]
@tomholub 
> You could cancel the background task when no pass phrase is available.

I see we already have that checking:

https://github.com/FlowCrypt/flowcrypt-android/blob/89f3f396d8501020013828044820174161a77be9/FlowCrypt/src/main/java/com/flowcrypt/email/service/actionqueue/actions/EncryptPrivateKeysIfNeededAction.kt#L55

So it happens only for weak pass phrases.

**Comment by tomholub** [comment]
> We can notify a user about a weak password and propose to change it (to make a backup with a much stronger password)

To keep it simple, you could just show a toast, and force view change to "change pass phrase" view

> As for me it'll better to make that in the next release. The current release alredy have a lot of changes

**Comment by DenBond7** [comment]
@tomholub 
> To keep it simple, you could just show a toast

What a text should I show?

**Comment by tomholub** [comment]
Your private key pass phrase is too weak. Please change it in the settings.

**Comment by DenBond7** [comment]
@tomholub 
> To keep it simple, you could just show a toast, and force view change to "change pass phrase" view

I've changed this behavior a little. I've decided to show notification instead of toast. 
The keys checking works in the background and I don't know where the user will be when that checking will be completed.
Now it looks like

![image](https://user-images.githubusercontent.com/2863246/63252894-61411f80-c279-11e9-8141-e3970b2c7733.png)

When the user click on that notification he will be moved to the "change pass phrase" screen (which will be added to the existing hierarchy)
